bolt-modules/boltlib/lib/puppet/functions/run_task.rb

```diff
@@ -57,16 +51,33 @@ def run_task_raw(task_name, targets, task_args = nil, &block)
 
     use_args = task_args.reject { |k, _| k.start_with?('_') }
 
-    task_signature.runnable_with?(use_args) do |mismatch_message|
-      raise with_stack(:TYPE_MISMATCH, mismatch_message)
-    end || (raise with_stack(:TYPE_MISMATCH, 'Task parameters do not match'))
+    # Don't bother loading the local task definition if the --no-local-validation
+    # command line option has been specified and all targets use the PCP protocol
+    if executor.local_validation == false && targets.all? { |t| t.protocol == 'pcp' }
```

Comment:
I think we can change the check here to:
`if targets.all? { |t| t.options[:"local-validation"] == false}`

That simplifies the executor and makes this transport option more consistent even if there isn't a good use case for local-validation on only some nodes.